python/cudf/cudf/core/frame.py

```diff
@@ -1501,7 +1516,13 @@ def _drop_na_columns(self, how="any", subset=None, thresh=None):
                 thresh = len(df)
 
         for col in self._data.names:
-            if (len(df[col]) - df[col].null_count) < thresh:
+            no_threshold_valid_count = (
+                (len(df[col]) - df[col].null_count) < thresh
+            ) or (
+                isinstance(df._data[col], cudf.core.column.NumericalColumn)
+                and ((len(df[col]) - df._data[col].nan_count) < thresh)
+            )
```

Comment:
May be better off doing a `nan --> null` here and then just checking the thresh once to avoid unnecessary synchronization.

Reply:
Done 👍